### PR TITLE
Update deployment pipelines 3 1 1

### PR DESCRIPTION
**Pull request by ljones140**
Updates deployment pipeline.

Important that prod is updated as we will now send App Version and Build Sha to docker image when built.
This will be returned on GET `https://api.clearlydefined.io/`

### CODE REVIEW

File: .github/workflows/build-and-deploy-prod.yml [hunk 1]
```diff
@@ -7,8 +7,21 @@ on:
...
+
+      - name: Upload package-lock.json
+        uses: actions/upload-artifact@v4
+        with:
+          name: package-lock.json
+          path: package-lock.json
+
   build-and-deploy-prod:
-    uses: clearlydefined/operations/.github/workflows/app-build-and-deploy.yml@v1.1.0
+    uses: clearlydefined/operations/.github/workflows/app-build-and-deploy.yml@v3.1.0
```
Comment (elrayle) [inline .github/workflows/build-and-deploy-prod.yml:24]:
```suggestion
        needs: upload-package-lock-json
        uses: clearlydefined/operations/.github/workflows/app-build-and-deploy.yml@v3.1.0
```

**Review by elrayle — COMMENTED** [review]
One suggestion, otherwise looks good.  Thanks for adding the artifact!

File: .github/workflows/build-and-deploy-prod.yml (hunk 1 above)
Comment (ljones140) [reply .github/workflows/build-and-deploy-prod.yml]: Done, thanks

**Review by elrayle — APPROVED** [review]
Looks good.  Thanks for bringing this workflow in line with the others.

**Pull request merged**